cpp/src/reductions/segmented/mean.cu

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "compound.cuh"
+
+#include <cudf/detail/segmented_reduction_functions.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+
+namespace cudf {
+namespace reduction {
+
+std::unique_ptr<cudf::column> segmented_mean(column_view const& col,
+                                             device_span<size_type const> offsets,
+                                             cudf::data_type const output_dtype,
+                                             null_policy null_handling,
+                                             rmm::cuda_stream_view stream,
+                                             rmm::mr::device_memory_resource* mr)
+{
+  using reducer = compound::detail::compound_segmented_dispatcher<cudf::reduction::op::mean>;
+  return cudf::type_dispatcher(col.type(),
+                               reducer{},
+                               col,
+                               offsets,
+                               output_dtype,
+                               null_handling,
+                               1,  // ddof is not used for mean
```

Comment:
Do you think it would be a worthwhile refactor to make the `ddof` parameter into a `std::optional`?

Reply:
It is a detail function which eventually must be resolved as an actual number when calling the compound-reduce kernel. That is, I would not want to add an if-statement to every device thread. So somewhere this has to happen. I have an issue with the comment itself and can reword it -- the hardcoded value `1` is correct for `mean` but a different hardcoded value may be required for some future compound reduction.